src/python/pants/backend/helm/util_rules/sources.py

```diff
@@ -102,15 +170,22 @@ async def get_helm_source_files(request: HelmChartSourceFilesRequest) -> HelmCha
             ),
         ),
         Get(
-            StrippedSourceFiles,
+            SourceFiles,
             SourceFilesRequest([request.field_set.sources], enable_codegen=False),
         ),
     )
+
+    stripped_source_files = await _strip_chart_source_root(source_files, source_root)
```

Comment:
This is still using source roots, and I think you shouldn't be any more right?

Reply:
the `SourceRoot` instance being used here comes from the `find_chart_source_root` rule, which returns a `SourceRoot` from a `HelmChartSourceRootRequest`. I could create a custom response type to make it clear that this is not the coming from the core rules and prevent confusion...